src/Microsoft.AspNetCore.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcCoreBuilderExtensions.cs

```diff
@@ -55,14 +77,39 @@ private static void AddViewComponentApplicationPartsProviders(ApplicationPartMan
                 throw new ArgumentNullException(nameof(setupAction));
             }
 
-            builder.AddDataAnnotations();
-            AddViewServices(builder.Services);
+            AddViews(builder);
+            builder.Services.Configure(setupAction);
 
-            if (setupAction != null)
```

Comment:
Teensy cleanup: Don't duplicate _part_ of `AddViews()` here and code a few lines up `null`-checks `setupAction`.